src/Relation/Binary/PropositionalEquality/Core.agda

```diff
@@ -66,6 +66,17 @@ respʳ _∼_ refl x∼y = x∼y
 resp₂ : ∀ (∼ : Rel A ℓ) → ∼ Respects₂ _≡_
 resp₂ _∼_ = respʳ _∼_ , respˡ _∼_
 
+------------------------------------------------------------------------
+-- Some variants of refl and cong; 'i' for implicit
+-- 'e' for explicit
+pattern erefl x = refl {x = x}
+
+irefl : ∀ {f : A → B} x → f x ≡ f x
```

Comment:
Just been having another think about this... I'm happy with `erefl` as it is the explicit version of `refl`. However I'm less keen on `irefl`, as according the comment above it stands for "implicit refl" which to my mind is what the existing `refl` already is (`refl : ∀ {x} → x ≡ x`).

Is it not actually another degenerate version of `icong`? It seems to me that we have two dimensions over `cong`. The first is whether the `f` is implicit or explicit, the second is whether the equality `x ≡ y` is present or not. Under this analysis we have:

`cong :  ∀ (f : A → B) {x y} → x ≡ y → f x ≡ f y`
`icong :  ∀ {f : A → B} {x y} → x ≡ y → f x ≡ f y`
`??? : ∀ (f : A → B) x → f x ≡ f x` (doesn't exist)
`??? : ∀ {f : A → B} x → f x ≡ f x` (currently `irefl`)

I'm not quite sure what to suggest for the last two names. Obviously `cong'` and `icong'` aren't very meaningful, but we have used them before for degenerate versions of functions...

Reply:
Note that the second `???` is not really a 'cong', as it does not have any equality input. ["congruence" maps equalities to equalities](https://en.wikipedia.org/wiki/Congruence_relation).  By the same reasoning, the first `???` isn't either.

Bikeshedding:
- `crefl`, for "equality of **c**ontexts around x"?
- ` refl₁` (where we started), as it abstracts out over 1 argument. We could generalize. The normal `refl` is the 0-ary version.

Reply:
> Obviously cong' and icong' aren't very meaningful, but we have used them before for degenerate versions of functions...

That was my reasoning too in https://github.com/agda/agda-stdlib/issues/1198#issuecomment-626394293

Reply:
The reasoning for the ' is clear. I'm not convinced by the reasoning for calling it `icong`. It's not a congruence!

Reply:
Hmm I guess I think of it as a congruence with the equality `x ≡ x` omitted.

Reply:
Fine. I much prefer having this in the library than not, so `icong` it is.